snmplib/mib.c

```diff
@@ -647,9 +647,15 @@ sprint_realloc_octet_string(u_char ** buf, size_t * buf_len,
     case NETSNMP_STRING_OUTPUT_GUESS:
         hex = 0;
         for (cp = var->val.string, x = 0; x < (int) var->val_len; x++, cp++) {
-            if (!isprint(*cp) && !isspace(*cp)) {
+#if defined(WIN32) /*RHG 2021.09.10: Windows includes non-ascii characters (above 0x7F) as printing, which can generate ugly strings */
```

Comment:
Please do not add any #ifdef WIN32 in code that should be generic. Additionally, this change is not related to NETSNMP_OID_OUTPUT_FULL_AND_NUMERIC and hence should not be in this patch.